From 9aa10a1c4f9bb235ebe1f09c27c565f5f5a2f270 Mon Sep 17 00:00:00 2001
From: "Yan, Xiangyang" <xiangyangx.yan@intel.com>
Date: Tue, 21 Mar 2017 13:31:33 +0800
Subject: [PATCH 0037/1240] trusty-ipc:tipc_msg_hdr structure: support large
 message transfer

len field type of tipc_msg_hdr structure is u16 which will only
 handle message length of less than 64K. Change it to u32 to support
larger message.

Change-Id: I9f08d699842723224a10242d19165fa748a8c8b4
Signed-off-by: Yan, Xiangyang <xiangyangx.yan@intel.com>
---
 drivers/trusty/trusty-ipc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/trusty/trusty-ipc.c b/drivers/trusty/trusty-ipc.c
index 363b023..44843eb 100644
--- a/drivers/trusty/trusty-ipc.c
+++ b/drivers/trusty/trusty-ipc.c
@@ -71,9 +71,9 @@ struct tipc_dev_config {
 struct tipc_msg_hdr {
 	u32 src;
 	u32 dst;
-	u32 reserved;
-	u16 len;
+	u32 len;
 	u16 flags;
+	u16 reserved;
 	u8 data[0];
 } __packed;
 
-- 
2.7.4

